scripts/publish-tree.ts

```diff
@@ -1,17 +1,105 @@
 import { program } from "commander";
+import dotenv from "dotenv";
+import axios from "axios";
+import FormData from "form-data";
+import fs from "fs";
 
-// TODO
-// Get inspired from: https://github.com/UMAprotocol/protocol/blob/master/packages/merkle-distributor/scripts/2_PublishClaimsForWindow.ts
+import { parseInputFile, writeToOutput } from "../src/fs-utils";
+import {
+  assertAndGetPinataEnvVars,
+  assertAndGetScraperEnvVars,
+  checkRecipientAmountsAndDuplicates
+} from "../src/validation";
+import { treeFileSchema } from "../src/schemas";
+
+dotenv.config();
 
 program
   .requiredOption(
     "-i, --input <path>",
     "input JSON file location containing output of 'yarn create-tree'"
   )
+  .option(
+    "-ss, --skip-scraper",
+    "optional flag whether to skip scraper api upload",
+    true // TODO: set to false if Scraper API endpoint ready
+  )
   .parse(process.argv);
 
 async function main() {
-  console.log("TODO");
+  const options = program.opts();
+
+  console.log(`Publishing ${options.input} to IPFS and Scraper API...`);
+
+  const rawInputFile = parseInputFile(options.input);
+  const inputFile = {
+    ...rawInputFile,
+    // We transform this to an array because yup doesn't have good support for
+    // schemas with dynamic keys and type inference
+    recipientsWithProofs: Object.values(rawInputFile.recipientsWithProofs)
+  };
+  const inputFileName = options.input.split("/").at(-1) as string;
```

Comment:
nit: maybe consider using [path.basename()](https://nodejs.org/docs/latest-v16.x/api/path.html#pathbasenamepath-suffix), as it's a little more standard and readable.

Reply:
nice that is indeed more readable. changed here https://github.com/across-protocol/merkle-distributor/pull/2/commits/12e061bfa00dcbb40457f1e811bb7d2191c4eda5